src/Database/Mysql/Driver/Mysql.php

```diff
@@ -54,15 +59,46 @@ public function update(Tenant $tenant): bool
             return false;
         }
 
-        return $this->process($tenant, [
-            'user' => "RENAME USER `{$config['oldUsername']}`@'{$config['host']}' TO `{$config['username']}`@'{$config['host']}'",
-        ]);
+        if (isset($config['allowedHost'])) {
+            $config['host'] = $config['allowedHost'];
+        }
+
+        $tempTenant = $tenant;
+        $tempTenant->{$tempTenant->getTenantKeyName()} = $tenant->getOriginal($tenant->getTenantKeyName());
+        $originalTenant = Tenancy::getTenant();
```

Comment:
Do we need some kind of Tenancy::temporarilyReplaceTenant() method or helper class to set and reset a tenant?

Reply:
It would be interesting, but I'm really afraid of unexpected behaviour/people not knowing what state their application is in.

Extra note:

I had a quick thought about it last night and was thinking of maybe creating a new connection that would be named with a random string or something like that.

Reply:
What if we made a function on Tenants that would provide us with a connection for that specific tenant?